libp2p/utility.nim

```diff
@@ -65,3 +65,23 @@ when defined(libp2p_agents_metrics):
   const
     KnownLibP2PAgents* {.strdefine.} = "nim-libp2p"
     KnownLibP2PAgentsSeq* = KnownLibP2PAgents.safeToLowerAscii().tryGet().split(",")
+
+template safeConvert*[T: SomeInteger, S: Ordinal](a: S): T =
```

Comment:
```suggestion
template safeConvert*[T: SomeInteger, S: Ordinal](a: S): T =
   # Converts `s` to `T` iff it can't trigger defects
```
Or something like that?

Reply:
It's not even about defects, this compiles, and prints 18446744073709551615
```nim
let i = -1
echo uint64(i)
```